src/rlookup.c

```diff
@@ -24,17 +24,50 @@ static RLookupKey *createNewKey(RLookup *lookup, const char *name, size_t name_l
   }
 
   // Set the name of the key.
-  ret->name = name;
+  ret->name = (flags & RLOOKUP_F_NAMEALLOC) ? rm_strndup(name, name_len) : name;
   ret->name_len = name_len;
-  ret->path = name;
+  ret->path = ret->name;
   ret->dstidx = lookup->rowlen;
+  ret->flags = flags & ~RLOOKUP_TRANSIENT_FLAGS;
 
   // Increase the RLookup table row length. (all rows have the same length).
   ++(lookup->rowlen);
 
   return ret;
 }
 
+// Allocate a new RLookupKey and add it to the RLookup table.
+static RLookupKey *overrideKey(RLookup *lk, RLookupKey *old, uint32_t flags) {
+  RLookupKey *new = rm_calloc(1, sizeof(*new));
+
+  // Copy the old key to the new one
+  new->name = old->name; // taking ownership of the name
+  new->name_len = old->name_len;
+  new->path = new->name; // keeping the default of path = name
+  new->dstidx = old->dstidx;
+
+  // Set the new flags
+  new->flags = flags & ~RLOOKUP_TRANSIENT_FLAGS;
+  // If the old key was allocated, we take ownership of the name.
+  new->flags |= old->flags & RLOOKUP_F_NAMEALLOC;
+
+  // Make the old key inaccessible for new lookups
+  old->flags &= (old->path == old->name) ? ~RLOOKUP_F_NAMEALLOC : UINT32_MAX; // If the old key allocated the name and not the path, we take ownership of the allocation
```

Comment:
either break into if
or
define new mask and call it (the mask value is UNIT32_MASK